django/db/models/query.py

```diff
@@ -338,7 +338,22 @@ def __or__(self, other):
     ####################################
 
     def _iterator(self, use_chunked_fetch, chunk_size):
```

Comment:
Looking back at it this can probably all be reduced to

```python
iterable = self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
if not self._prefetch_related_lookups:
    yield from iterable
    return

iterator = iter(iterable)
while True:
    results = list(islice(iterator, chunk_size))
    if not results:
        break
    prefetch_related_objects(results, *self._prefetch_related_lookups)
    yield from iter(results)